specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json

```diff
@@ -2474,6 +2474,128 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/sqlDatabases/{databaseName}/containers/{containerName}/throughputSettings/default/retrieveThroughputDistribution": {
+      "post": {
+        "operationId": "SqlResources_SqlContainerRetrieveThroughputDistribution",
+        "x-ms-examples": {
+          "CosmosDBSqlContainerRetrieveThroughputDistribution": {
+            "$ref": "./examples/CosmosDBSqlContainerRetrieveThroughputDistribution.json"
+          }
+        },
+        "description": "Retrieve throughput distribution for an Azure Cosmos DB SQL container",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseNameParameter"
+          },
+          {
+            "$ref": "#/parameters/containerNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "retrieveThroughputParameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RetrieveThroughputParameters"
+            },
+            "description": "The parameters to provide for retrieving throughput distribution for the current SQL container."
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The retrieve throughput distribution for SQL container operation will complete asynchronously."
+          },
+          "200": {
```

Comment:
>200 [](http://example.com/codeflow?start=11&length=3)

Does this API sometimes complete synchronously and sometimes async? ARM guidance is that each API should always do one or the other. So if you support async, you don't need 200. Same feedback applies to all 4 of the new POST APIs. Ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously)

Reply:
It behaves the same way as is described in the link you shared. It returns 202 when the retrieve throughput operation is accepted. Once it has gotten provisioned throughputs from all physical partitions it returns that in the response with a 200. It is modelled similar to the partitionMerge operation that got introduced in this preview version

Reply:
The link I shared has the 200 coming from a different endpoint that the client polls with GET, not the same endpoint that returns the 202 from POST. So how does it work to sometimes get 200 and sometimes get 202 from this this API? How does the API decide whether to synchronously return the last completed result with 200 or to start a new async request and return 202?

Reply:
makes sense. I have updated the spec with examples